libraries/rush-lib/src/api/PnpmLockValidationConfiguration.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { AlreadyReportedError, FileSystem, JsonFile, JsonSchema } from '@rushstack/node-core-library';
+import fs from 'fs';
+import yaml from 'js-yaml';
+import type { Lockfile } from '@pnpm/lockfile-types';
+import type { ITerminal } from '@rushstack/terminal';
+
+import schemaJson from '../schemas/pnpm-lock-validation.schema.json';
+import type { RushConfiguration } from './RushConfiguration';
+import { CustomTipId } from './CustomTipsConfiguration';
+
+export interface IPnpmValidationJson {
+  pnpmLockValidationRules?: Array<{
+    ruleName: 'validateIntegritySha1';
+    description: string;
+    pattern: string;
```

Comment:
If the purpose of this rule is to prevent SHA1 hashes, why should the user need to provide a RegExp?  It should be Rush's job to ensure that SHA1 is eliminated, and if the lockfile format changes, it should be Rush's responsibility to improve the RegExp.

On the other hand, we could instead consider a more flexible rule like `forbidRegExpInLockfile`, however flexibility is often a worse user experience.  Where possible, it's better for Rush to _**solve a problem well**_ rather than merely _**enabling users to solve the problem themselves.**_ The reason is that solving problems _**well**_ tends to be a lot of work. 😁